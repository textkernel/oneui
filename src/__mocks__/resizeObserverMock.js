export class ResizeObserverMock {
    observe = jest.fn();

    unobserve = jest.fn();

    disconnect = jest.fn();
}
