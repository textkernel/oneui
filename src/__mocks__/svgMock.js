/* eslint-disable react/display-name, react/destructuring-assignment */
import React from 'react';

module.exports = (props) => <svg data-testid={props['data-testid'] || 'default-icon'} />;
