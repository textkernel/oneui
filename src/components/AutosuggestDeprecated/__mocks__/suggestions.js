export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptune' },
    { name: 'Pluto', disabled: true },
];

export const SUGGESTION_TO_STRING = (item) => (item ? item.name : '');
