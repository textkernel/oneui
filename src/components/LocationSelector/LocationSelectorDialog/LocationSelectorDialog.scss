$input-line-height: 67px;
$location-card-width: 300px;
$slider-indication-width: 90px;

.LocationSelectorDialog {
    &__inputLine {
        display: flex;
        box-sizing: border-box;
        height: $input-line-height;
        padding: var(--space-150);
    }

    &__searchField {
        flex-basis: 80%;
        min-width: 100px;
    }
    
    &__slider {
        width: 50%;
        max-width: 400px;
        flex-direction: row;
        display: flex;
        margin: var(--space-200) 5px var(--space-200) 25px;
    }

    &__sliderLabel {
        width: $slider-indication-width;
        margin: 0;
        text-align: right;
    }

    &__button {
        flex-basis: 20%;
        margin-left: var(--space-150);
        height: 38px;
    }

    &__locationsWrapper {
        display: flex;
        height: calc(100% - #{$input-line-height});
    }

    &__locationCardsContainer {
        padding: 0;
        margin: 0;
        width: $location-card-width;
        min-width: $location-card-width;
        padding-right: calc(var(--space-150) - 1px);
        overflow: auto;
        height: 100%;
    }

    &__locationCard:not(:last-child) {
        margin-bottom: var(--space-50);
    }
}
