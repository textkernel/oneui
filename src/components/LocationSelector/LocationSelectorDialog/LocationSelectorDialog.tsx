import * as React from 'react';
import { bem } from '../../../utils';
import { LocationCard, LocationCardProps } from '../../LocationCard';
import { Text } from '../../Text';
import { Slider } from '../../Sliders';
import { Button } from '../../Buttons';
import { LocationAutocomplete, LocationAutocompleteProps } from '../../LocationAutocomplete';
import { Map, CircularMarker } from '../../Map';
import { SIZES, ENTER_KEY } from '../../../constants';
import styles from './LocationSelectorDialog.scss';
import { LocationSelectorLocation } from '../utils';

// Props for child components that we just push through with prop-drilling
type CardPropsForSelectorDialog = Pick<
    LocationCardProps,
    | 'locationId'
    | 'hasRadius'
    | 'sliderLabel'
    | 'minRadius'
    | 'maxRadius'
    | 'radiusStep'
    | 'onRadiusChange'
>;

type AutocompletePropsForSelectorDialog = Pick<
    LocationAutocompleteProps,
    | 'inputPlaceholder'
    | 'noSuggestionsPlaceholder'
    | 'isFocused'
    | 'clearTooltipLabel'
    | 'country'
    | 'placeTypes'
    | 'showCountryInSuggestions'
    | 'onRemoveAllLocations'
>;

interface Props extends CardPropsForSelectorDialog, AutocompletePropsForSelectorDialog {
    /** defines if location cards should be rendered or not */
    withoutLocationCards?: boolean;
    /** stores an array of selected location objects */
    selectedLocations: LocationSelectorLocation[];
    /** radius label renderer e.g. radius => `+ ${radius} km` */
    renderRadiusLabel: (radius: number) => string;
    /** address to make initial map centering more specific */
    initialMapAddress?: string;
    /** function to be executed if error occurs while fetching suggestions */
    onLocationAutocompleteError?: (status: google.maps.places.PlacesServiceStatus) => void; // LocationAutocompleteProps.onError
    /** label for the Done button */
    doneLabel: string;
    /** function called with location object as an argument when it is selected from the suggestions */
    onAddLocation: (location: LocationSelectorLocation) => void; // LocationAutocompleteProps.onSelectionChange
    /** function called with a location details as an argument to be changed */
    onUpdateLocation: (locationId: string | undefined, radius: number) => void; // LocationCardProps.onRadiusChange
    /** function with a locationId as an argument to be removed */
    onRemoveLocation: (locationId: string | undefined) => void; // LocationCardProps.onDelete
    /** function to calculate marker positions in Map  */
    getMarkers?: () => CircularMarker[];
    /** function to be called when teh Done button is clicked */
    onCloseModal?: () => void;
    /** A geoJson description of the area that should be highlighted when there are no other markers present */
    defaultHighlight?: GeoJSON.GeoJsonObject; // MapProps.defaultHighlight
}

const { elem } = bem('LocationSelectorDialog', styles);

const LocationSelectorDialog: React.FC<Props> = ({
    /** FieldWrapper props */
    inputPlaceholder,

    /** LocationCard props */
    hasRadius,
    minRadius,
    maxRadius,
    radiusStep,
    renderRadiusLabel,
    onRemoveLocation,
    doneLabel,
    clearTooltipLabel,

    /** LocationAutocomplete props */
    country,
    initialMapAddress = null,
    defaultHighlight = undefined,
    placeTypes,
    noSuggestionsPlaceholder,
    showCountryInSuggestions,
    onLocationAutocompleteError = null,

    /** Internal use */
    withoutLocationCards = false,
    onUpdateLocation,
    selectedLocations,
    getMarkers = () => undefined,
    onAddLocation,
    onRemoveAllLocations,
    onCloseModal = () => null,
}) => {
    const locationInputRef = React.createRef<HTMLInputElement>();

    const [firstSelectedLocation] = selectedLocations;

    function getDefaultArea() {
        if (initialMapAddress || country) {
            return { address: initialMapAddress || country };
        }

        return undefined;
    }

    function handleAddLocation(location) {
        if (locationInputRef.current && !withoutLocationCards) {
            setTimeout(() => locationInputRef.current?.focus());
        }
        onAddLocation(location);
    }

    function handleRemoveLocation(locationId) {
        if (locationInputRef.current) {
            locationInputRef.current.focus();
        }
        onRemoveLocation(locationId);
    }

    function handleRadiusChange(radius) {
        onUpdateLocation(firstSelectedLocation.id, radius);
    }

    function handleInputFormSubmit(e) {
        if (e.key === ENTER_KEY) {
            e.stopPropagation();
            onCloseModal?.();
        }
    }

    return (
        <>
            <div {...elem('inputLine')} role="presentation" onKeyDown={handleInputFormSubmit}>
                <LocationAutocomplete
                    {...elem('searchField')}
                    isFocused
                    inputRef={locationInputRef}
                    defaultInputValue={
                        withoutLocationCards && firstSelectedLocation
                            ? firstSelectedLocation.description
                            : ''
                    }
                    inputPlaceholder={inputPlaceholder}
                    clearTooltipLabel={clearTooltipLabel}
                    noSuggestionsPlaceholder={noSuggestionsPlaceholder}
                    onSelectionChange={handleAddLocation}
                    country={country}
                    placeTypes={placeTypes}
                    singleLocation={withoutLocationCards}
                    showCountryInSuggestions={showCountryInSuggestions}
                    onRemoveAllLocations={onRemoveAllLocations}
                    onError={onLocationAutocompleteError}
                    hidePoweredByGoogleLogo
                />
                {hasRadius && withoutLocationCards && selectedLocations.length === 1 && (
                    <div {...elem('slider')}>
                        <Slider
                            value={firstSelectedLocation.radius}
                            min={minRadius}
                            max={maxRadius}
                            step={radiusStep}
                            railStyle={{ backgroundColor: 'var(--color-neutral-25)' }}
                            onChange={handleRadiusChange}
                        />
                        <Text size={SIZES[0]} {...elem('sliderLabel')}>
                            {renderRadiusLabel(firstSelectedLocation.radius)}
                        </Text>
                    </div>
                )}
                <Button {...elem('button')} onClick={onCloseModal} isPrimary>
                    {doneLabel}
                </Button>
            </div>
            <div {...elem('locationsWrapper')}>
                {!withoutLocationCards && selectedLocations.length > 0 && (
                    <ul {...elem('locationCardsContainer')}>
                        {selectedLocations.map((location) => (
                            <LocationCard
                                {...elem('locationCard')}
                                As="li"
                                key={location.id}
                                locationId={location.id}
                                locationTitle={location.description}
                                distanceRadius={location.radius}
                                sliderLabel={renderRadiusLabel(location.radius)}
                                hasRadius={hasRadius}
                                minRadius={minRadius}
                                maxRadius={maxRadius}
                                radiusStep={radiusStep}
                                onRadiusChange={onUpdateLocation}
                                onDelete={handleRemoveLocation}
                            />
                        ))}
                    </ul>
                )}
                <Map
                    defaultArea={getDefaultArea()}
                    markers={getMarkers?.()}
                    defaultHighlight={defaultHighlight}
                />
            </div>
        </>
    );
};

LocationSelectorDialog.displayName = 'LocationSelectorDialog';

export { LocationSelectorDialog, Props as LocationSelectorDialogProps };
