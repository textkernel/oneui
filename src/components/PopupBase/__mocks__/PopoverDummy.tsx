import * as React from 'react';
import { Button } from '../../..';

interface Props {
    setPopupVisibility: (isVisible) => void;
}
export const PopoverDummy = React.forwardRef<HTMLDivElement, Props>(
    ({ setPopupVisibility }, ref) => {
        const popupStyle = {
            backgroundColor: '#FFEEDD',
            padding: '10px',
            borderRadius: '5px',
            border: '1px solid #CCC',
            boxShadow: '0 3px 6px 6px rgba(0, 0, 0, 0.05)',
        };
        return (
            <div style={popupStyle} ref={ref} role="group">
                <p>I am the content of pop up.</p>
                <p>Can be a form or anything.</p>
                <Button onClick={() => setPopupVisibility(false)}>Close me!</Button>
            </div>
        );
    }
);

PopoverDummy.displayName = 'Popover';
