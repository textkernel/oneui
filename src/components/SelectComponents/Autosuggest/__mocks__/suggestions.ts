export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptune' },
    { name: 'Pluto', disabled: true },
];

export type SuggestionsType = (typeof SUGGESTIONS)[number];

export const COMPLEX_SUGGESTIONS = [
    { name: 'Sun', type: 'star' },
    { name: 'Moon', type: 'star' },
    { name: 'Mercury', type: 'planet' },
    { name: 'Venus', type: 'planet' },
    { name: 'Earth', type: 'planet' },
    { name: 'Mars', type: 'planet' },
    { name: 'Jupiter', type: 'planet' },
    { name: 'Saturn', type: 'planet' },
    { name: 'Neptune', type: 'planet' },
];

export const SUGGESTION_TO_STRING = (item) => (item ? item.name : '');
