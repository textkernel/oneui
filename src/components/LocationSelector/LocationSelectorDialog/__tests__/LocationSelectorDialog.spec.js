import React from 'react';
import toJson from 'enzyme-to-json';
import { act } from 'react-dom/test-utils';
import { stabGoogleApi, getPlacePredictionsMock } from '../../../../__mocks__/googleApiMock';
import predictionsMock from '../../../LocationAutocomplete/__mocks__/predictions.json';
import { LocationSelectorDialog } from '../LocationSelectorDialog';

stabGoogleApi();

describe('LocationSelectorDialog component', () => {
    const selectedLocations = [
        {
            id: 'ajdo-219a-j19v-0491',
            description: 'Amsterdam',
            center: {
                lng: 4.894539799999961,
                lat: 52.3666969,
            },
            radius: 42,
            sliderLabel: '42km',
        },
        {
            id: 'ajdo-219a-j19v-0492',
            description: 'Utrecht',
            center: {
                lng: 5.121420100000023,
                lat: 52.09073739999999,
            },
            radius: 20,
            sliderLabel: '20km',
        },
    ];

    const minRadius = 1;
    const maxRadius = 100;
    const radiusStep = 1;
    const renderRadiusLabel = (r) => `+ ${r} km`;
    const country = 'NL';
    const placeTypes = ['(regions)'];

    const inputPlaceholder = 'inputPlaceholder';
    const noSuggestionsPlaceholder = 'noSuggestionsPlaceholder';
    const doneLabel = 'doneLabel';

    const onAddLocationMock = jest.fn();
    const onUpdateLocationMock = jest.fn();
    const onRemoveLocationMock = jest.fn();
    const onRemoveAllLocationsMock = jest.fn();
    const onLocationAutocompleteErrorMock = jest.fn();
    const onCloseModal = jest.fn();

    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <LocationSelectorDialog
                inputPlaceholder={inputPlaceholder}
                minRadius={minRadius}
                maxRadius={maxRadius}
                radiusStep={radiusStep}
                renderRadiusLabel={renderRadiusLabel}
                onRemoveLocation={onRemoveLocationMock}
                doneLabel={doneLabel}
                country={country}
                placeTypes={placeTypes}
                noSuggestionsPlaceholder={noSuggestionsPlaceholder}
                showCountryInSuggestions
                onLocationAutocompleteError={onLocationAutocompleteErrorMock}
                onUpdateLocation={onUpdateLocationMock}
                selectedLocations={selectedLocations}
                getMarkers={() => []}
                onAddLocation={onAddLocationMock}
                onCloseModal={onCloseModal}
                onRemoveAllLocations={onRemoveAllLocationsMock}
            />
        );
    });

    afterEach(() => {
        jest.resetAllMocks();
    });

    it('should render correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should close LocationSelectorDialog by pressing on Done button', () => {
        wrapper.find('.Button').simulate('click');
        expect(onCloseModal).toHaveBeenCalledTimes(1);
    });

    it('should call onAddLocation by selecting an item from the autosuggestion list', () => {
        jest.useFakeTimers();
        getPlacePredictionsMock.mockImplementationOnce((req, cb) => cb(predictionsMock, 'OK'));
        wrapper.find('input').simulate('change', { target: { value: 'Tonga' } });
        act(() => {
            jest.runAllTimers();
        });
        wrapper.find('input').simulate('click');

        expect(wrapper.find('Autosuggest').find('li')).toHaveLength(5);
        expect(onAddLocationMock).not.toHaveBeenCalled();

        // wrapper.find('Autosuggest').find('li').at(0).childAt(0).simulate('click');

        // expect(onAddLocationMock).toHaveBeenCalledTimes(1);
    });

    it('should call onRemoveLocation by clicking on Close button of the selected location item', () => {
        expect(onRemoveLocationMock).not.toHaveBeenCalled();

        wrapper.find('ul').at(1).childAt(0).find('button').simulate('click');

        expect(onRemoveLocationMock).toHaveBeenCalledTimes(1);
    });

    // TODO: Update test so slider handle would be reachable in order to update a LocationCard
    it.skip('should call onUpdateLocation by setting a new location radius valueof the selected location item', () => {
        expect(onUpdateLocationMock).not.toHaveBeenCalled();

        wrapper
            .find('ul')
            .at(1)
            .childAt(0)
            .find('div.rc-slider-handle')
            .simulate('mouseover')
            .simulate('mousedown')
            .simulate('mousemove')
            .simulate('click');

        expect(onUpdateLocationMock).toHaveBeenCalled();
    });
});
