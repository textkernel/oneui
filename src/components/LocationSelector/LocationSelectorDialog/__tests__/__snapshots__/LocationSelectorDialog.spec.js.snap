// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSelectorDialog component should render correctly 1`] = `
<LocationSelectorDialog
  country="NL"
  doneLabel="doneLabel"
  getMarkers={[Function]}
  initialMapAddress={null}
  inputPlaceholder="inputPlaceholder"
  maxRadius={100}
  minRadius={1}
  noSuggestionsPlaceholder="noSuggestionsPlaceholder"
  onAddLocation={[MockFunction]}
  onCloseModal={[MockFunction]}
  onLocationAutocompleteError={[MockFunction]}
  onRemoveAllLocations={[MockFunction]}
  onRemoveLocation={[MockFunction]}
  onUpdateLocation={[MockFunction]}
  placeTypes={
    Array [
      "(regions)",
    ]
  }
  radiusStep={1}
  renderRadiusLabel={[Function]}
  selectedLocations={
    Array [
      Object {
        "center": Object {
          "lat": 52.3666969,
          "lng": 4.894539799999961,
        },
        "description": "Amsterdam",
        "id": "ajdo-219a-j19v-0491",
        "radius": 42,
        "sliderLabel": "42km",
      },
      Object {
        "center": Object {
          "lat": 52.09073739999999,
          "lng": 5.121420100000023,
        },
        "description": "Utrecht",
        "id": "ajdo-219a-j19v-0492",
        "radius": 20,
        "sliderLabel": "20km",
      },
    ]
  }
  showCountryInSuggestions={true}
  withoutLocationCards={false}
>
  <div
    className="LocationSelectorDialog__inputLine"
    onKeyDown={[Function]}
    role="presentation"
  >
    <LocationAutocomplete
      className="LocationSelectorDialog__searchField"
      clearLabel=""
      country="NL"
      defaultInputValue=""
      hidePoweredByGoogleLogo={true}
      inputPlaceholder="inputPlaceholder"
      inputRef={
        Object {
          "current": <input
            aria-autocomplete="list"
            aria-labelledby="downshift-0-label"
            autocomplete="off"
            class="AutosuggestDeprecated__input"
            data-lpignore="true"
            id="downshift-0-input"
            placeholder="inputPlaceholder"
            value=""
          />,
        }
      }
      isFocused={true}
      noSuggestionsPlaceholder="noSuggestionsPlaceholder"
      onError={[MockFunction]}
      onRemoveAllLocations={[MockFunction]}
      onSelectionChange={[Function]}
      placeTypes={
        Array [
          "(regions)",
        ]
      }
      showCountryInSuggestions={true}
      singleLocation={false}
    >
      <AutosuggestDeprecated
        className="LocationSelectorDialog__searchField"
        clearTitle=""
        defaultInputValue=""
        getSuggestions={Array []}
        iconNode={
          <FaMapMarkerAlt
            className="LocationAutocomplete__icon"
          />
        }
        inputPlaceholder="inputPlaceholder"
        inputRef={
          Object {
            "current": <input
              aria-autocomplete="list"
              aria-labelledby="downshift-0-label"
              autocomplete="off"
              class="AutosuggestDeprecated__input"
              data-lpignore="true"
              id="downshift-0-input"
              placeholder="inputPlaceholder"
              value=""
            />,
          }
        }
        isFocused={true}
        isLoading={false}
        isMultiselect={false}
        isProminent={false}
        listRef={null}
        listRenderer={[Function]}
        noSuggestionsPlaceholder="noSuggestionsPlaceholder"
        onBlur={[Function]}
        onClearAllSelected={[MockFunction]}
        onInputValueChange={[Function]}
        onSelectionChange={[Function]}
        rootRef={null}
        saveSelectedValueToInput={false}
        selectedPlaceholder=""
        selectedSuggestions={null}
        showClearButton={false}
        suggestionToString={[Function]}
      >
        <div
          className="AutosuggestDeprecated LocationSelectorDialog__searchField"
        >
          <Downshift
            defaultHighlightedIndex={0}
            defaultIsOpen={false}
            environment={[Window]}
            getA11yStatusMessage={[Function]}
            inputValue=""
            itemToString={[Function]}
            onChange={[Function]}
            onInputValueChange={[Function]}
            onOuterClick={[Function]}
            onSelect={[Function]}
            onStateChange={[Function]}
            onUserAction={[Function]}
            scrollIntoView={[Function]}
            selectedItemChanged={[Function]}
            stateReducer={[Function]}
            suppressRefError={false}
          >
            <div
              aria-expanded={false}
              aria-haspopup="listbox"
              aria-labelledby="downshift-0-label"
              aria-owns={null}
              className="AutosuggestDeprecated__main AutosuggestDeprecated__main--focused"
              role="combobox"
            >
              <FieldWrapper
                className="AutosuggestDeprecated__field"
                clearLabel=""
                disabled={false}
                isArrowUp={false}
                isFocused={true}
                onClear={[Function]}
                onClick={[Function]}
                onKeyDown={[Function]}
                showArrow={false}
                showClearButton={false}
              >
                <div
                  className="FieldWrapper FieldWrapper--isFocused AutosuggestDeprecated__field"
                  disabled={false}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                >
                  <div
                    className="FieldWrapper__content"
                  >
                    <div
                      className="AutosuggestDeprecated__wrapper AutosuggestDeprecated__wrapper--focused"
                      role="searchbox"
                      tabIndex="0"
                    >
                      <FaMapMarkerAlt
                        className="AutosuggestDeprecated__spacedElem LocationAutocomplete__icon"
                      >
                        <IconBase
                          attr={
                            Object {
                              "viewBox": "0 0 384 512",
                            }
                          }
                          className="AutosuggestDeprecated__spacedElem LocationAutocomplete__icon"
                        >
                          <svg
                            className="AutosuggestDeprecated__spacedElem LocationAutocomplete__icon"
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 384 512"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                              key="0"
                            />
                          </svg>
                        </IconBase>
                      </FaMapMarkerAlt>
                      <input
                        aria-activedescendant={null}
                        aria-autocomplete="list"
                        aria-controls={null}
                        aria-labelledby="downshift-0-label"
                        autoComplete="off"
                        className="AutosuggestDeprecated__input"
                        data-lpignore={true}
                        id="downshift-0-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        placeholder="inputPlaceholder"
                        value=""
                      />
                    </div>
                    <List
                      aria-labelledby="downshift-0-label"
                      className="AutosuggestDeprecated__list--focused"
                      doSelectOnNavigate={false}
                      id="downshift-0-menu"
                      isControlledNavigation={true}
                      isDivided={false}
                      role="listbox"
                    >
                      <ul
                        aria-labelledby="downshift-0-label"
                        className="List AutosuggestDeprecated__list--focused"
                        id="downshift-0-menu"
                        role="listbox"
                      />
                    </List>
                  </div>
                </div>
              </FieldWrapper>
            </div>
          </Downshift>
        </div>
      </AutosuggestDeprecated>
    </LocationAutocomplete>
    <Button
      className="LocationSelectorDialog__button"
      context="brand"
      disabled={false}
      isBlock={false}
      isInline={false}
      onClick={[MockFunction]}
      size="normal"
      type="button"
    >
      <button
        className="Button Button--context_brand LocationSelectorDialog__button"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        doneLabel
      </button>
    </Button>
  </div>
  <div
    className="LocationSelectorDialog__locationsWrapper"
  >
    <ul
      className="LocationSelectorDialog__locationCardsContainer"
    >
      <LocationCard
        As="li"
        className="LocationSelectorDialog__locationCard"
        distanceRadius={42}
        hasRadius={false}
        key="ajdo-219a-j19v-0491"
        locationId="ajdo-219a-j19v-0491"
        locationTitle="Amsterdam"
        maxRadius={100}
        minRadius={1}
        onDelete={[Function]}
        onRadiusChange={[MockFunction]}
        radiusStep={1}
        sliderLabel="+ 42 km"
      >
        <li
          className="LocationCard LocationSelectorDialog__locationCard"
        >
          <div
            className="LocationCard__header"
          >
            <Text
              className="LocationCard__title"
              context="default"
              inline={false}
              size="large"
              title="Amsterdam"
            >
              <p
                className="Text--size_large LocationCard__title"
                title="Amsterdam"
              >
                Amsterdam
              </p>
            </Text>
            <button
              className="LocationCard__deleteButton"
              onClick={[Function]}
              type="button"
            >
              ×
            </button>
          </div>
        </li>
      </LocationCard>
      <LocationCard
        As="li"
        className="LocationSelectorDialog__locationCard"
        distanceRadius={20}
        hasRadius={false}
        key="ajdo-219a-j19v-0492"
        locationId="ajdo-219a-j19v-0492"
        locationTitle="Utrecht"
        maxRadius={100}
        minRadius={1}
        onDelete={[Function]}
        onRadiusChange={[MockFunction]}
        radiusStep={1}
        sliderLabel="+ 20 km"
      >
        <li
          className="LocationCard LocationSelectorDialog__locationCard"
        >
          <div
            className="LocationCard__header"
          >
            <Text
              className="LocationCard__title"
              context="default"
              inline={false}
              size="large"
              title="Utrecht"
            >
              <p
                className="Text--size_large LocationCard__title"
                title="Utrecht"
              >
                Utrecht
              </p>
            </Text>
            <button
              className="LocationCard__deleteButton"
              onClick={[Function]}
              type="button"
            >
              ×
            </button>
          </div>
        </li>
      </LocationCard>
    </ul>
    <Map
      defaultArea={
        Object {
          "address": "NL",
        }
      }
      mapContainerStyle={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
      markers={Array []}
    >
      <GoogleMap
        mapContainerStyle={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        onLoad={[Function]}
        options={
          Object {
            "fullscreenControl": false,
            "mapTypeControl": false,
            "rotateControl": false,
            "streetViewControl": false,
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </GoogleMap>
    </Map>
  </div>
</LocationSelectorDialog>
`;
